### ISSUE TITLE
bpnet or per nucleotide output handling in tangermeme?

### ISSUE THREAD

**Issue by DavidvanBruggen** [jmschrei/tangermeme#4]
Thank you for the wonderful package!

I was wondering if you could perhaps show a tutorial how a BPNet like output is handled in your deeplift/shap function?
As currently it seems that it requires a target that is then only specifying the index on the last dimension of the prediction?
So it seems like it would default to showing attributions for index 0? So the first letter in the one hot matrix? How could this be extended to one attribution value to the whole output?

Thanks!

**Comment by jmschrei** [comment]
Hi David

When using gradient-based methods like DeepLIFT, you need to specify a single output to "explain." BPNet has two outputs: the profile, and the log read count. The log read count, being a single number per example, can easily be explained. The profile is a bit trickier because, as you note, it's actually 1000 numbers. The standard way to explain the profile is to actually explain `y.dot(softmax(y))` where `y` is the predicted logits.  In my other package, [bpnet-lite](https://github.com/jmschrei/bpnet-lite) I have this implemented as a set of two wrapper: [CountWrapper](https://github.com/jmschrei/bpnet-lite/blob/master/bpnetlite/bpnet.py#L113) which allows you to calculate attributions against the count head, and [ProfileWrapper](https://github.com/jmschrei/bpnet-lite/blob/master/bpnetlite/bpnet.py#L85), which allows you to calculate attributions using this profile aggregation strategy. 

Let me know if you have any other questions

**Comment by jmschrei** [comment]
Please re-open if issues persist.
